samples/guestbook/inject_aws.go

```diff
@@ -60,16 +62,19 @@ func awsBucket(ctx context.Context, cp awsclient.ConfigProvider, flags *cliFlags
 	return b, func() { b.Close() }, nil
 }
 
-// awsSQLParams is a Wire provider function that returns the RDS SQL connection
-// parameters based on the command-line flags. Other providers inside
-// awscloud.AWS use the parameters to construct a *sql.DB.
-func awsSQLParams(flags *cliFlags) *rdsmysql.Params {
-	return &rdsmysql.Params{
-		Endpoint: flags.dbHost,
-		Database: flags.dbName,
-		User:     flags.dbUser,
-		Password: flags.dbPassword,
+// openAWSDatabase is a Wire provider function that connects to RDS based on
+// the command-line flags.
+func openAWSDatabase(ctx context.Context, opener *rdsmysql.URLOpener, flags *cliFlags) (*sql.DB, func(), error) {
```

Comment:
Should this function call the portable open url function so that it looks clear since the parameter is a url string?

Reply:
Yes, but I there's a larger cleanup issue open for that: #1676